includes/class.llms.generator.php

```diff
@@ -1188,6 +1435,83 @@ public function set_generator( $generator = null ) {
 
 	}
 
+	/**
+	 * Sideload an image from a url
+	 *
+	 * @since [version]
+	 *
+	 * @param int    $post_id WP_Post ID of the post where the image will be attached.
+	 * @param string $url     The image's URL.
+	 * @return string|int|WP_Error Returns a WP_Error on failure, the image's new URL when `$return` is "src", otherwise returns the image's attachement ID.
+	 */
+	protected function sideload_image( $post_id, $url, $return = 'src' ) {
+
+		// Check if the image was previously sideloaded.
+		$id = empty( $this->images[ $url ] ) ? false : $this->images[ $url ];
+
+		// Image was not previously sideloaded.
+		if ( ! $id ) {
+
+			$id = media_sideload_image( $url, $post_id, null, 'id' );
+			if ( is_wp_error( $id ) ) {
+				return $id;
+			}
+
+			// Store the ID for future usage.
+			$this->images[ $url ] = $id;
+
+		}
+
+		return 'src' === $return ? wp_get_attachment_url( $id ) : $id;
+
+	}
+
+	/**
+	 * Sideload images found in a given post
+	 *
+	 * This attempts to sideload the `src` attribute of every <img> element
+	 * found in the `post_content` of the supplied post.
+	 *
+	 * @since [version]
+	 *
+	 * @param LLMS_Post_Model $post Post object.
+	 * @return null|boolean Returns `true` on success, `false` if there were no images to update, or `null` if sideloading is disabled.
+	 */
+	protected function sideload_images( $post ) {
+
+		// Sideloading is disabled.
+		if ( ! $this->is_image_sideloading_enabled() ) {
+			return null;
+		}
+
+		$content  = $post->post->post_content;
```

Comment:
```suggestion
		$content  = $post->get( 'content', true );
```
just to use our own api :D

Reply:
Fine.